\documentclass[10pt,a4paper]{article}
\usepackage[top=0.85in,footskip=0.75in]{geometry}

%
\usepackage{amsmath,amssymb}

%
\usepackage{changepage}

%
\usepackage[utf8x]{inputenc}

%
\usepackage{textcomp,marvosym}

%
%\usepackage{cite}

%
\usepackage{nameref,hyperref}

%
\usepackage[right]{lineno}
\linenumbers %
\usepackage{gensymb}

%
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

%
%\usepackage[table]{xcolor}

\usepackage{mathtools}

%
\usepackage{array}

%
\usepackage{float}

%
\newcolumntype{+}{!{\vrule width 2pt}}

%
\newlength\savedwidth
\newcommand\thickcline[1]{%
	\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
	\cline{#1}%
	\noalign{\vskip\arrayrulewidth}%
	\noalign{\global\arrayrulewidth\savedwidth}%
}

%
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
	\hline
	\noalign{\global\arrayrulewidth\savedwidth}}


%
\usepackage{setspace}
%

\usepackage[dvipsnames]{xcolor}


%
\usepackage[hang,aboveskip=5pt,labelfont=bf,labelsep=period,justification=justified,singlelinecheck=off,textfont=small]{caption}
\renewcommand{\figurename}{Fig.}

%\bibliographystyle{naturemag}

\usepackage[numbers,sort&compress]{natbib}
\bibliographystyle{unsrt}
%
%

%
%\makeatletter
%\renewcommand{\@biblabel}[1]{\quad#1.}
%\makeatother

%
\date{}

%
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%
%
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[LR]{1.00in}
\fancyfootoffset[LR]{1.00in}
%

%


\renewcommand{\a}{{\bf a}}
\renewcommand{\b}{{\bf b}}

\newcommand{\ol}{\overline}
%
\newcommand{\ep}{\epsilon}
\newcommand{\s}{\sigma}
\renewcommand{\L}{{\cal L}}
\newcommand{\F}{{\cal F}}
\newcommand{\comment}[1]{{\noindent \color{blue} #1}}
\newcommand{\corr}[1]{{\noindent \color{red} \small #1}}
\newcommand{\add}[1]{{\noindent \color{green} \small #1}}
\newcommand{\purple}[1]{\textcolor{purple}{#1}}
\newcommand{\outcomment}[1]{}
%
\newcommand{\EQ}{\begin{equation}}
\newcommand{\EE}{\end{equation}}
\newcommand{\EQA}{\begin{eqnarray}}
\newcommand{\EEA}{\end{eqnarray}}
\newcommand{\productlog}{\mathrm{ProductLog}}

%
\graphicspath{ {../figures/} }
\newcounter{firstbib}
%


\begin{document}
	\vspace*{0.2in}

	%
	\begin{flushleft}
		{\Large
			\textbf\newline{Survival of the complex/Survival of the most flexible}
		}
		\newline
		\\
		Tom R\"oschinger\textsuperscript{1},
		Simone Pompei\textsuperscript{2},
		Michael L\"assig\textsuperscript{3,*},
		\\
		\bigskip
		\textbf{1} Division of Chemistry and Chemical Engineering, California Institute of Technology, Pasadena, CA 91125 \\
		\textbf{2} \\
		\textbf{3} Institut f\"ur Biologische Physik, Universit\"at zu K\"oln,
		Z\"ulpicherstr. 77, 50937, K\"oln, Germany
		\\
		\bigskip

		%
		%
		%
		%
		%
		%\textbf{\P} These authors contributed equally to this work.

		%
		* mlaessig@uni-koeln.de

	\end{flushleft}
	
	
	\subsection*{Abstract}
	

	\subsection*{Introduction}
	The complexity of a system is a crucial component in molecular evolution. Evolution of gene regulation is often assumed to be driven by the gain and loss of transcription factor binding sites (TFBS), rather than the evolution of a transcription factor (TF) itself \cite{tugrul_dynamics_2015}. This assumption is based on the pleotropy of most TFs that have been observed, and that the pleotropic effects of a mutation in such a transcription factor are deleterious for most interactions, such that no change is possible. However, these TF's are conserved across species to begin with \cite{schmidt_five-vertebrate_2010}. Some classes of TFs display extensive levels of species diversity while maintaining structure and function \cite{nowick_lineage-specific_2010} and it has been discussed that protein evolution is a crucial source of developmental variation \cite{lynch_resurrecting_2008}, but the consequences of protein evolution remain unclear \cite{wagner_gene_2008, voordeckers_how_2015}. The evolution of gene regulatory networks is assumed to be driven by modifications 
	and the evolution of TFs \cite{lozada-chavez_bacterial_2006, perez_evolution_2009}.
	
	\subsection*{Methods}
	\input{methods.tex}

	

	\subsection*{Results}
	\subsubsection*{Evolutionary Steady State in Non-equilibrium}
	{\color{ForestGreen} This section should contain all necessary information and needs work on the language.}\\
	We start by investigating how externally driven mutations that change the consensus sequence of the binding site force the system into a new state that is different from the steady state that can be observed when binding is only disrupted by mutations in the binding site. Therefore, we use the Fokker-Planck equation describing the evolution of the mean $\Gamma$ of a quantitative trait $E$ \cite{nourmohammad_evolution_2013},
	\begin{equation}
		\frac{\partial}{\partial t}Q(\Gamma,t)=\left[\frac{\tilde{g}^{\Gamma\Gamma}}{2N}\frac{\partial^2}{\partial\Gamma^2}- \frac{\partial}{\partial\Gamma}\left(m^\Gamma+\tilde{g}^{\Gamma\Gamma}\tilde{s}_\Gamma\right)\right]Q(\Gamma,t)\label{equ:trait_diffusion}.
	\end{equation}
	The equation can be broken down into three evolutionary forces: genetic drift, selection and mutations. In the low mutation rate regime, the selection term can be written as $\tilde{s}_\Gamma = \partial_\Gamma f(\Gamma)$, i.e., the derivative of the fitness landscape at the value of the trait mean. The effective diffusion coefficient is given by $\tilde g^{\Gamma\Gamma} = \langle \Delta\rangle \sim \mu$ (\purple{include scaling with alphabet size here?}), hence scaling the strength of selection with the variation of the trait within a population (\purple{mention Fisher's theorem here?}). 
	The mutational drift term is $m^\Gamma = -\frac{n}{n-1} (\Gamma-\Gamma_0)$, where $n$ is the alphabet size of the binding site and $\Gamma_0$ is the mean trait in the absence of selection. 
	The mutation term describes deterministic drift that sites undergo due to mutations. If this term dominates the deterministic behavior, then most sites will have binding energies similar to random sequences. Driver mutations have the same impact on the binding energy as trailer mutations. Hence, in the evolutionary description of traits, we can add the driver mutation rate $\rho$ to the trailer mutation rate $\mu$ in the mutation term, $m^\Gamma\sim-(\rho+\mu)(\Gamma-\Gamma_0)$, for details see \textcolor{blue}{methods oder supplementary}.\\
	The selection term $\tilde{g}^{\Gamma\Gamma}\tilde{s}_\Gamma$ describes the selection coefficient of the trait mean value $\tilde{s}_\Gamma$ and how it is amplified by the diversity within a population $\tilde{g}^{\Gamma\Gamma}=\langle\Delta_E\rangle $. For low mutation rates, the diversity within a population is given by its value in the absence of selection, and very small. Most positions are monomorphic, thus a mutation in the driver sequence changes the binding energy for nearly all sites equally. Therefore, the diversity within a population is not affected by driver mutations, and the selection term is invariant.\\
	The non-deterministic term $\tilde{g}^{\Gamma\Gamma}/2N$ describes how mutations lead to a spread in mean binding energies across populations. In the original formulation this term also scales with the mean diversity within a population. However, driver mutations also lead to an increased diversity across populations, since they change the mean binding energy, while not effecting the mean diversity within a population. Hence, we have to distinguish between the two response terms $\tilde{g}^{\Gamma\Gamma}$. The response coefficient in the selection term, which only scales with the trailer mutation rate $\mu$, is labeled $\tilde{g}^{\Gamma\Gamma}_s\sim\mu$. The response coefficient in the non-deterministic term is scaling with both mutation rates and is labeled $\tilde{g}^{\Gamma\Gamma}_d\sim\mu+\rho$.\\
	In the absence of selection, the steady state distribution $Q_0(\Gamma)$ is the same as in the absence of driving, since the scaling of both terms cancels out in the final solution (\textcolor{blue}{Methods}). However, in a driven environment $\rho>0$, the steady state solution becomes
	\begin{equation}
		Q(\Gamma)=\frac{1}{Z}Q_0(\Gamma)\exp\left[\frac{2N}{1+\kappa}F(\Gamma)\right],\label{equ:steady_state}
	\end{equation}
	where $\kappa=\rho/\mu$ is the ration of driver and trailer mutation rate, which is the key parameter describing the strength of non-equilibrium. A rescaling of the fitness landscape $\hat{F}(\Gamma)=F(\Gamma)/(1+\kappa)$ reproduces the steady state distribution in equilibrium with a weaker fitness landscape. This result is explained by the different scalings of the population genetic terms with non-equilibrium. The mutation drift towards smaller binding energies increases as well as diffusion across populations, while selection is not increased and therefore relatively weaker.\\
	Note, that the rescaling of the fitness landscape effectively only applies to the term in the fitness landscape regarding the binding probability. The fitness length cost is absorbed in the normalization of the steady state distribution in \ref{equ:steady_state}.
	In the low mutation rate limit, an exact steady state distribution can be numerically computed from substitution rates (Methods or Supplementary). In the regime of small selection coefficients $Ns\sim1$, an analytic steady state can be approximated (Methods or Supplementary), which is equal to the result in equation \ref{equ:steady_state}.

	\begin{figure}
		\includegraphics[width=\linewidth]{figure1_concept.pdf}\caption{\color{ForestGreen} Conceptual figure. Left side should show various distributions that evolved for fixed lengths at different fitness parameters $f_0$ and how they change with increasing $\kappa$. For now, I simply plotted histograms, and the colors in each plot are the different fitness parameters (as seen in the legend on the right plot). Then, to show the distributions vary, we could show the mean (or peak) of each distribution on the right side and how it changes with increasing $\kappa$. This simulation was run \textbf{without recovery} of sites at the lower plateau, but is being repeated right now with recovery. This figure would support the first claim that the distribution in non-equilibrium can be written as the steady-state distribution with and additional factor of $1/(1+\kappa)$ in the exponent. Axis labels are going to be fixed in the next iteration.}\label{fig1}
	\end{figure}


	\subsubsection*{Optimal Binding Site Length}
	In this section we discuss how the genetic load can be minimized by adapting the length of the binding site in non-equilibrium. The genetic load is defined as the difference of the mean fitness of the population to the maximum achievable fitness. (\purple{Do we need a figure here?}). By minimzing genetic load a population maximizes its fitness. In general increasing the length of genetic element comes with a fitness cost, since more base pairs need to be kept specific to maintain function. Therefore, we include a linear fitness cost to the fitness landscape (Methods). Note, this additional fitness cost does not change dynamics when binding site lengths are kept constant because the term cancels when computing selection coefficients.
	Non-equilibrium weakens the ability to find specific binding sites, while the length fitness cost is independent of the state of the binding. Hence, the fitness trade-off changes with the level of non-equilibrium. We can consider the fitness effects in the terms of genetic load. For weak non-equilibrium most sites are functional and are at the edge of the upper fitness plateau, which can approximated by an exponential tail. Then the genetic load of a population is given by $\mathcal{L}\sim \left|\frac{\partial}{\partial \Gamma} f(\Gamma,l)\right|+f_l\,l$. The average load across populations is given by the load at the deterministic mutation-selection balance $\hat{\Gamma}$(Methods or Supplementary) \cite{held_survival_2019}. We retrieve two terms for the genetic load, with different scaling of the binding length. The linear fitness cost for binding length, and the binding probability term with scales inversely with binding length,
	\begin{equation}
		\mathcal{L} = \xi \frac{l_0}{l}(1+\kappa)+\lambda \, \frac{l}{l_0} = \mathcal{L}_\kappa + \mathcal{L}_\lambda,
		\label{equ:maintext_load}
	\end{equation}
	Where $\xi$ is a combination of various parameters, $\lambda=2Nf_l/l_0\sim 1$ is the scaled fitness cost for length,  and $l_0$ is a length scaling parameter the derivation of this equation can be found in the supplementary methods. Due to the trade off between the cost of adding positions to the binding site and reducing load by adding positions, there is an optimal binding site length, which minimizes the total genetic load,
	\begin{equation}
		l_\mathrm{opt}\sim l_0\sqrt{\frac{1+\kappa}{\lambda}}.
	\end{equation}
	Hence, we obtain a direct scaling of the optimal binding site length with non-equilibrium. The scaled fitness length cost in the model is chosen such that the binding site length in equilibrium ($\kappa=0$) is ten, $l_\mathrm{opt}(\kappa)=0$, which is the typical length of transcription factor binding sites \cite{stewart_why_2012}. 


\subsubsection*{Dynamical Evolution of Binding Site Length}
{\color{ForestGreen} This section should be very descriptive, since it is one of the main results. And why considering the ratchet is important when writing down models of this kind which could consider binding site length. Also, could this be a reason for larger genomes, as soon as selection pressure decreases? Also, this section should have a minimal number of equations, just tell the story, and any computation should be in the supplementary methods, to make this section as straightforward to understand as possible. }\\

In a real population, length adaptation is a dynamical process, driven by mutations which include a new position in the sequence in its recognition. We assume these mutations to be very rare compared to other type of mutations, such that the timescales separate. Therefore, we can assume that the system is in the respective steady state of binding energies each time a length mutation occurs. Assuming an alphabet size of $n$ and using the match/mismatch energy model (see Methods), a newly added position will be a match with probability 1/4, since it has not been selected for yet and hence is random compared to the driver sequence. Length increase mutations are on average neutral in respect to the adaption term in the fitness landscape in first order, and are only influenced by the fitness cost of length (supplementary methods),
\begin{equation}
	\sigma^+ = -\frac{\lambda}{l_0} =-\frac{\partial \mathcal{L}_\lambda}{\partial l}.
\end{equation}
Since $\lambda\sim 1$, length increase mutations are in the regime of weak selection (\purple{Here either an explanation of what this means or a nice reference}).
However, length decrease mutations are affected by the fact that the binding site is adapted, i.e., removing a position is likely to hit an interaction which is beneficial to the binding interaction. If the binding site has $k$ matches, then the probability of removing a match is $k/l$, which is much larger than the probability of adding a match, which we showed to be $1/4$. The selection coefficient for removing a position has an additional term,
\begin{equation}
	\sigma^- = \frac{\lambda}{l_0} - \xi \epsilon\beta \left(\frac{l_0}{l}\right)^2 (1+\kappa) = \frac{\partial \mathcal{L}_\lambda}{\partial l} + l_0 \epsilon\beta\frac{\partial \mathcal{L_\kappa}}{\partial l}.
\end{equation}
Unexpectedly, the adaptation term is scaled by the effective length scaling parameter, and the mutational effect. Therefore, the selection coefficient for length decrease mutations shows marginal selection, $\sigma^-\sim (l/l_0)^2\sim 1$. Due to the separation of time scales, we can write down substitution rates for length mutations, which lead to a steady state distribution. In first order, we can compute the substitution rates as 
\begin{equation}
	u_{+/-} \sim 1 + \frac{\sigma^{+/-}}{2} \sim \exp\left[\frac{\sigma^{+/-}}{2} \right].
\end{equation}
Using detailed balance (\purple{Maybe avoid this term here?}), we can compute the steady state distribution for binding site lengths. Using the rates above, we find that the resulting distribution is
\begin{equation}
	P(l) \simeq \exp\left[-\frac{\epsilon\beta l_0}{2}\mathcal{L}_\kappa - \mathcal{L}_\lambda\right].
	\label{equ:length_distribution}
\end{equation}
When comparing the maximum of this distribution with the length with minimizes the load, we find that the length $l_opt^*$ which optimizes the effective potential resulting from the dynamical calculation is larger by a factor $\sqrt{l_0}$,
\begin{equation}
	l_{opt}^* = \epsilon\beta\sqrt{l_0}\,l_{opt}.
\end{equation}
This is an enormous difference, see Figure~(\purple{insert figure}), and shows how important it is to consider the dynamics of the evolution of binding site lengths. Transcription factor binding sites have lengths around 10bp in prokaryotes (\purple{cite "Why are TF binding sites 10bp long."}), so we can tune the fitness cost for length $f_l$ such that the optimal length computed from the dynamical computation is around that value in equilibrium ($\kappa=0$). Then, we observe an increase in binding site length with increasing non-equilibrium, due to the sites being under more selective pressure to maintain their binding specificity, which shifts the weights in the trade off with the cost of increased binding site length.

\subsubsection*{Data}
\textcolor{ForestGreen}{Placeholder for a possible section on some data. This has the lowest priority so far, and I think we should finish everything but the discussion before we dive into this.}


\subsection*{Discussion}
\textcolor{ForestGreen}{Here we should discuss the meaning of our results. The two factors of increased binding site length. First, the dynamics of length evolution lead to a much higher binding site length in equilibrium, than observed from minimizing the genetic load. But. Also we should discuss which data could support our theory. And finally, the shortcoming of the theory. Like how the approximation of the distribution is not really good for higher selection coefficients. And what how it could be used to improve. And finally how we think experiments could be set up to test the hypthesis experimentally.}



\bibliography{complexity_refs.bib}


\section*{Acknowledgments}


\section*{Author Contributions}


\section*{Competing Interests}
The authors declare no competing interests.

\clearpage
\subsection*{To Do List}
	
\begin{itemize}
	\color{ForestGreen}
	\item Urgent:
	\begin{itemize}
		\item Michael: Supplementary Methods 3: Does the diversity within a population depend on the alphabet size, as seen from the analytic results? $\Delta E ~\sim \left(\frac{n}{n-1}\right)^2$. This is important for the results in this section. Either mathematical proof or logical explanation.
		\item Tom: Think about loss at various fitness parameters, and how that relates to observable non-equilibrium.
		\item Tom: Get a first draft of the methods section ready so it is easy to refer to it when writing the results. The methods should not change too much anymore, unless we include data analysis, but this can be added later.
		\item Tom: Add conceptual figures (Optimal length scaling with non-equilibrium, effect of ratchet, two dimensional distribution $Q(\gamma, l)$ moving in 2D space with increasing $\kappa$).
	\end{itemize}
	\item Remaining Tasks:
	\begin{itemize}
		\item Get all necessary simulations done and store results with README.md files to remember what they where.
		\item Finalize analytical computations in supplementary methods.
		\item Formulate punchline of the paper.
		\item Check references.
		\item Write introduction based on the results.
		\item Finalize results section.
		\item Spend \~ one month looking for possible data sets.
		\item Either add data analysis or write discussion and be done.
	\end{itemize}
\end{itemize}


\newpage 
\comment{
\paragraph{Comment 29. 7., edited 8. 9. Does the slow dynamics of $\ell$ lead to a steady state that is localized at $\ell_{\rm opt}$?}
\begin{enumerate}
\item Our model has a conditional stationary distribution $Q(k|\ell)$ peaked at $k/\ell \equiv \gamma$ with
\EQ
\gamma - \gamma_0 = \frac{\ell_0}{\ell}
\EE
and $\gamma_0 = 1/4$. This stationary state has a scaled load $\L \equiv (F - F_{\max})/ \tilde \sigma$ given by
\EQ
\L = \L_\kappa + \L_\lambda = (\gamma - \gamma_0) (1 + \kappa) + \lambda  \frac{\ell}{\ell_0},
\EE
leading to an optimum length
\EQ
\ell_{\rm opt} = \ell_0 \sqrt{\frac{1+\kappa}{\lambda}}.
\EE

\item In the exponential regime of the (scaled) fitness landscape $\F(k |\ell) \equiv F(k|\ell) / \tilde \sigma$, which is relevant for stable sites, the scaled load equals the scaled selection coefficient of site mutations,
\EQ
s = \frac{\partial \F(k,\ell)}{\partial k} = \L_\kappa = (\gamma - \gamma_0) (1 + \kappa);
\EE
prefactors to be double-checked with Torsten-Daniel paper.

\item Perturbation theory in $\ep \equiv 1/\ell_0$. We can distinguish terms of order $\ep$ and of order $\ell/\ell_0 \sim 1$. We have $\gamma \sim \lambda \sim 1$ and $\L \sim 1$; the scaling of $\lambda$ follows from the requirement $\ell_{\rm opt} \approx \ell_0$ at $\kappa = 0$. In particular, we can distinguish selection coefficients $ \sim 1/\ell_0$ (weak selection) and $ \sim 1$ (marginal selection).

\item Adiabatic substitution dynamics of $\ell$. Length changes are assumed to occur with much smaller rates than mutations; we assume they are emitted from the steady state of the $k$ dynamics. Each length change is coupled to a stochastic change of $k$. Because the dynamics takes place in the regime of weak and marginal selection, we use a linear approximation for substitution probabilities. Hence, we can first compute average selection coefficients $\bar s_+$ and $\bar s_-$ for changes of $\ell$ and then evaluate the corresponding substitution rates using these averages.

In this framework, we find the processes $\ell \to \ell +1$ with average selection coefficient
\EQA
\bar s_+ & = & \frac{3}{4} \frac{(-s)}{4} + \frac{1}{4} \frac{3s}{4} - \frac{\lambda}{\ell_0}
\\
& = & - \frac{\lambda}{\ell_0}
\EEA
(near-neutral evolution) and the processes $\ell \to \ell - 1$ with average scaled selection coefficent
\EQA
\bar s_- & = & \gamma \frac{(-3s)}{4} + (1 - \gamma) \frac{s}{4} + \frac{\lambda}{\ell_0}
\\
& = & -(\gamma - \gamma_0) s + \frac{\lambda}{\ell_0}
\\
& = & - (\gamma - \gamma_0)^2 (1 + \kappa) + \frac{\lambda}{\ell_0}
\\
& = & \ell_0 \frac{\partial \L_\kappa}{\partial \ell} + \frac{\partial \L_\lambda}{\partial \ell}
\EEA
(marginal selection). These selection coefficients are of different magnitude: length increases are constrained only by the weak linear term, length decreases are marginally constrained by conservation of site function.

\item These asymmetric dynamics define an adaptive ratchet, which acts to increase selection for complexity. A typical cycle $\ell \to \ell + 1 \to \ell$ has the following average scaled fitness balance:
\begin{itemize}
\item substitution $\ell \to \ell + 1$: weak fitness loss, $\bar s_+ = - \lambda / \ell_0 = - O(\ep)$;
\item equilibration at $\ell + 1$: weak fitness gain, $\Delta \L_{\ell + 1} = \L_\kappa (\ell) - \L_\kappa (\ell +1) \sim \ell_0 /\ell^2 = O( \ep)$;
\item substitution $\ell + 1 \to \ell$: marginal fitness loss,   $\bar s_- = - \ell_0^2 /\ell^2 =- O(\ep^0) $;
\item equilibration at $\ell$: marginal fitness gain, $\Delta \L_{\ell} = - \L_\kappa (\ell) + \L_\kappa (\ell +1) - \bar s_- = O(\ep^0)  $.
\end{itemize}
This cycle could inform a Figure.

\item Effective fitness landscape for length changes. We compute the ratio of the forward and backward substitution rate in first-order approximation,
\EQA
\frac{u_{\ell \to \ell+1}}{u_{\ell + 1 \to \ell}} & = & \frac{\exp(\bar s_+/2)}{\exp(\bar s_-/2)} +O(s^2)
\\
& \simeq & \exp \left [ - \frac{\ell_0}{2} (\L_\kappa (\ell+1) - \L_\kappa (\ell)) - (\L_\lambda (\ell+1) - \L_\lambda (\ell)) \right ]
\\
& \equiv & \exp \big [ \F_{\rm eff} (\ell+1) - \F_{\rm eff} (\ell) \big ]
\label{u_ratio}
\EEA
with an effective scaled fitness potential
\EQ
\F_{\rm eff} (\ell) = - \frac{\ell_0}{2} \L_\kappa (\ell) - L_\lambda (\ell) .
 \EE
These rates define an equilibrium distribution
\EQ
Q(\ell) \sim \exp [ \F_{\rm eff} (\ell)],
\EE
which is peaked at
\EQ
\ell^* = \ell_0 \sqrt{ \frac{\ell_0}{2} \frac{(1+\kappa)}{\lambda}} = \ell_{\rm opt} \sqrt{ \frac{\ell_0}{2}} .
\EE
Compared to the naive distribution $Q(\ell) \sim \exp [ - \L (\ell)]$, which would lead to a peak at $\ell_{\rm opt}$, the ratchet mechanism enhances the evolution of complexity.

\item Summary.
\begin{itemize}
\item Non-equilibrium weakens the selection on point mutations at a given complexity, as expressed by an effective fitness landscape
\EQ
\F_{\rm eff} (k|\ell) \sim \frac{1}{1 + \kappa} .
\EE
\item Non-equilibrium introduces ratchet selection for complexity, as expressed by an effective fitness landscape
\EQ
\F_{\kappa, \rm eff} (\ell) \sim \ell_0(1+ \kappa).
\EE
This landscape is proportional to $\ell_0$; that is, complexity begets more complexity.
\end{itemize}

\item Questions:\\
(1) Is this in qualitative accordance with the evaluation of the rates in (\ref{u_ratio}) beyond first order?
\\
(2) How does the fitness balance compare with the numerics?
\\
(3) How does $\ell^*$ compare with the numerics?
\end{enumerate}


\paragraph{Comments 17. 12.}
\begin{enumerate}
\item We could map the performance of the ratchet -- in particular, the maximum complexity gain -- as a function of the following independent parameters (see notebook adaptive ratchet 3): 

\begin{itemize}
\item Binding free energy gap $\Delta G$ (in units of $k_B T$). Reasonable values: $\Delta G \gtrsim 6$. 

\item Functional binding length $\ell_0$ (excess number of binding positions per site compared to average random sequence),  
\EQ
\ell_0 = \frac{\Delta G}{\ep_0}, 
\EE
where $\ep_0 = \Delta \Delta G$ is the average free energy effect per mutation. Reasonable values $\ep_0 \sim 1$ (in units of $k_B T$). Smaller $\ep_0$ decrease curvature effects in the fitness landscape. 

\item Fitness effect per site, $f_0$. Reasonable values: $f_0 \gtrsim 10$. Smaller values of $f_0$ increase the probability of function loss (in particular, at high $\kappa$), larger values increase curvature effects in the fitness landscape. 
\end{itemize}

The constraint selection coefficient $\lambda$ is not independent, but is to be tuned so that equilibrium sites are close to the minimal functional length (to be implemented in notebook). 

\item We know that sites tend to lose function at high $\kappa$ and insufficient $f_0$. Can we analytically capture this effect by a condition for stability analogous to eq. (10) of the phenotypic interference paper? This would read 
\EQ
\frac{{\cal L}^*_b (\ell)}{\tilde \sigma} \lesssim \frac{f_0}{2}
\EE
with 
\EQA
\frac{{\cal L}^*_b (\ell)}{\tilde \sigma} & \equiv & \frac{F_b (\gamma \!=\! 1, \ell) - F_b (\gamma^*(\ell), \ell)}{\tilde \sigma} 
\nonumber 
\\
& = & \frac{(1 + \kappa)(\gamma^* (\ell) - \gamma_0)}{\ep_0} \simeq \frac{(1 + \kappa)}{\ep_0} \frac{\ell_0}{\ell}. 
\EEA

\item A preliminary observation from the notebook is that the complexity gain appears to remain capped at about 2, even when the sites remain functional at the ML point. This seems to be caused by the length-increasing mutations becoming more negatively selected at higher values of $\kappa$. Can we understand why and maybe overcome this limit in some parameter regime? 

\item Calibration of $\lambda$. Imposing the condition that the minimum functional site length equals the equilibrium length appears to give
\EQ
\lambda = \frac{1}{\ep_0} \frac{\ell_0^2}{\ell_{\rm min }^2 }
\EE
with $\ell_{\rm min} (\gamma_0, \ep_0)$ given in the notebook. This expression is numerically too small and needs to be refined. 
\end{enumerate}


}

\clearpage
\section*{Supplementary Information}

\paragraph*{Supplementary Methods~1}
\label{supp1}
\textbf{Steady State Approximations}\\

\noindent In the regime of low mutation rates, $\mu N\ll1$, a population is in a monomorphic state, i.e., dominated by a single genotype, 
for most of the time. A new mutation fixes in the population with probability $p(s)$ given by \cite{kimura_probability_1962},
\begin{equation}
p(s) = \frac{1-e^{-2s}}{1-e^{-2Ns}},
\end{equation}
where $s$ is the selection coefficient of the mutation and $N$ the effective population size. Then, substitutions occur at a rate
\begin{equation}
u(s)_{E\rightarrow E'} = N\mu_{E\rightarrow E'}p(s),
\end{equation}
where $\mu_{E\rightarrow E'}$ is the rate at which a mutation occurs which changing the binding energy from $E$ to $E'$. Here, we assumed that the fixation 
of a mutation is under selective pressure, which is the case for mutations in the trailer. We investigate the effects of mutations in the driver, which occur 
with rate $\rho$. These mutations are results of external events, e.g., a mutation in the coding sequence of a transcription factor leading to a missense mutation. 
In isolated system of driver and trailer, these mutations are not under selection pressure, but can change the binding energy nonetheless. 
Here we assume that these mutations happen for all driver sequences at the same time and therefore can be treated as substitutions as well. 
Thus, we add a term to the mutation rate in the form of
\begin{equation}
u(s)_{E\rightarrow E'}=   N\mu_{E\rightarrow E'}p(s) + \rho_{E\rightarrow E'},
\end{equation}
where $\mu_{E\rightarrow E'}$ is the mutation rate with driver mutations change the binding energy from $E$ to $E'$.
Assume, both trailer and driver have the same alphabet size, then both mutation rates are given by a constant rate times a factor accounting for 
the entropic difference between both states, $\mu_{E\rightarrow E'}=\mu\,w_{E\rightarrow E'}$ and $\rho_{E\rightarrow E'}=\rho\,w_{E\rightarrow E'}$. 
Even if the assumption of equal alphabet sizes doesn't hold, e.g., in protein-DNA interaction, the differences in alphabet sizes can be accounted for 
by rescaling the mutation rates by a constant factor. The entropic term $w$ is determined by the substitution rates in the absence of selection.
We are looking for the distribution of binding energies at steady state, which is equivalent to 
\begin{equation}
\frac{u(0)_{E\rightarrow E'}}{u(0)_{E'\rightarrow E}} = \frac{\mu_{E\rightarrow E'}}{\mu_{E'\rightarrow E}}= \frac{w_{E\rightarrow E'}}{w_{E'\rightarrow E}} = 
\frac{Q_0(E')}{Q_0(E)},
\end{equation}
where $Q_0(E)$ is the distribution of binding energies in the absence of fitness. Even though there is non-equilibrium on the level of genotypes, 
we can look for a steady state distribution of the binding energies. Since the dynamics are one dimensional and fully described by the substitution rates, 
we can find the steady state distribution $Q(k)$ by imposing detailed balance and solving
\begin{equation}
\frac{Q(E')}{Q(E)}=\frac{u(s)_{E\rightarrow E'}}{u(-s)_{E'\rightarrow E}}=  \frac{ N\mu_{E\rightarrow E'}p(s)+\rho_{E\rightarrow E'}}{N\mu_{E\rightarrow E'}p(s)
+\rho_{E\rightarrow E'}}.
\end{equation}
In equilibrium, $\rho=0$, the fraction has an exact solution \cite{berg_adaptive_2004},
\begin{equation}
	Q(k) = Q_0(k)\exp[2NF(k)].
	\label{equ:sub_equ_dist}
\end{equation}
In non-equilibrium $\rho>0$, the fraction does not simplify to an exact term as equation~\ref{equ:sub_equ_dist}, but we can compute the exact steady state 
distribution numerically. However, in the limit of small selection coefficients $Ns\sim1$, the can expand the fraction in orders of the selection coefficient 
and retrieve,
\begin{align}
\frac{Q_0(E')}{Q_0(E)}\frac{ N\mu p(s)+\rho}{N\mu p(s)+\rho}&\approx \frac{Q_0(E')}{Q_0(E)}\left[1+\frac{2s(N-1)}{1+\frac{\rho}{\mu}}+\frac{2s^2(N-1)^2}
{(1+\frac{\rho}{\mu})^2}+\mathcal{O}(s^3)\right] ,\nonumber\\
&\approx \frac{Q_0(E')}{Q_0(E)}\exp\left[\frac{2N}{1+\kappa}F(E)\right].
\end{align}
Populations are usually large, so $N-1\approx N$, and up to second order the series is equal to an exponential function. The resulting distribution is therefore 
given by
\begin{equation}
	Q(E) = Q_0(E) \exp\left[ \frac{2N}{1+\kappa} F(E) \right].
\end{equation}


\clearpage
\paragraph*{Supplementary Methods~2}
\label{supp2}
\input{supp2.tex}
\clearpage

\paragraph*{Supplementary Methods~3}
\label{supp3}
\input{supp3.tex}
\clearpage

\paragraph*{Supplementary Methods~4} 
\label{supp4}
\input{supp4.tex}



\end{document}
